sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java

```diff
@@ -283,8 +283,13 @@ private void sendRequestsForBulk(CompletableFuture<BitSet> bulkResultFuture, Map
                 @Override
                 public void onResponse(ShardResponse shardResponse) {
                     nodeJobsCounter.decrement(shardLocation.nodeId);
-                    processShardResponse(shardResponse);
-                    countdown();
+                    Exception failure = shardResponse.failure();
+                    if (failure != null && failure instanceof InterruptedException) {
+                        bulkResultFuture.completeExceptionally(failure);
```

Comment:
You're right, but that makes me thinking if this code is even required. In case of a kill the underlying BatchIterator will start throwing a failure on moveNext anyway, so I think we could ignore the InterruptException here and just let the BackpressureExecutor die - kill is best effort anyway